frontend/src/cljs/imas_seamap/map/events.cljs

```diff
@@ -556,24 +556,49 @@
                  [:ui.catalogue/catalogue-add-nodes-to-layer category layer "cat" [:data_classification]]
                  [:map/pan-to-layer layer]])})
 
-(defn update-active-network [db [_ network]]
-  (if-not (= network (get-in db [:map :active-network]))
-    (-> db
-        (assoc-in [:map :active-park] nil)
-        (assoc-in [:map :active-network] network))
-    db))
-
-(defn update-active-park [{:keys [map] :as db} [_ {:keys [network] :as park}]]
-  (-> db
-      (assoc-in [:map :active-network] (first-where #(= (:name %) network) (:networks map)))
-      (assoc-in [:map :active-park] park)))
-
-(defn update-active-zone [db [_ zone]]
-  (-> db
-      (assoc-in [:map :active-zone] zone)
-      (assoc-in [:map :active-zone-iucn] nil)))
-
-(defn update-active-zone-iucn [db [_ zone-iucn]]
-  (-> db
-      (assoc-in [:map :active-zone-iucn] zone-iucn)
-      (assoc-in [:map :active-zone] nil)))
+(defn update-active-network [{:keys [db]} [_ network]]
+  (let [db (if-not (= network (get-in db [:map :active-network]))
+             (-> db
+                 (assoc-in [:map :active-park] nil)
+                 (assoc-in [:map :active-network] network))
+             db)]
+    {:db db
+     :dispatch [:map/get-habitat-statistics]}))
+
+(defn update-active-park [{:keys [db]} [_ {:keys [network] :as park}]]
+  (let [db (-> db
+               (assoc-in [:map :active-network] (first-where #(= (:name %) network) (get-in db [:map :networks])))
+               (assoc-in [:map :active-park] park))]
+    {:db db
+     :dispatch [:map/get-habitat-statistics]}))
+
+(defn update-active-zone [{:keys [db]} [_ zone]]
+  (let [db (-> db
+               (assoc-in [:map :active-zone] zone)
+               (assoc-in [:map :active-zone-iucn] nil))]
+    {:db db
+     :dispatch [:map/get-habitat-statistics]}))
+
+(defn update-active-zone-iucn [{:keys [db]} [_ zone-iucn]]
+  (let [db (-> db
+               (assoc-in [:map :active-zone-iucn] zone-iucn)
+               (assoc-in [:map :active-zone] nil))]
+    {:db db
+     :dispatch [:map/get-habitat-statistics]}))
+
+(defn get-habitat-statistics [{:keys [db]}]
+  (let [habitat-statistics-url (get-in db [:config :habitat-statistics-url])
+        {:keys [active-network active-park active-zone active-zone-iucn]}                    (:map db)]
+   {:db db
```

Comment:
It might not be obvious, but you don't actually _have_ to return a :db key here.  The :db is just another effect that says "swap out the app-db and replace it with this one", and if it isn't present the app-db will stay as it is)

Reply:
I actually only noticed this feature today. Anyway, I'll fix it.